MSAL/src/requests/MSALBaseRequest.m

```diff
@@ -164,9 +165,79 @@ - (void)resolveEndpoints:(MSALAuthorityCompletion)completionBlock
 
 - (void)acquireToken:(nonnull MSALCompletionBlock)completionBlock
 {
-    NSMutableDictionary<NSString *, NSString *> *reqParameters = [NSMutableDictionary new];
+    MSIDTokenRequest *authRequest = [self tokenRequest];
     
-    // TODO: Remove once uid+utid work hits PROD
+    [authRequest sendWithBlock:^(id response, NSError *error) {
+        
+        [authRequest finishAndInvalidate];
+        
+        if (error)
+        {
+            completionBlock(nil, error);
+            return;
+        }
+        
+        if(response && ![response isKindOfClass:[NSMutableDictionary class]])
+        {
+            NSError *localError = CREATE_MSID_LOG_ERROR(_parameters, MSALErrorInternal, @"response is not of the expected type: NSMutableDictionary.");
+            completionBlock(nil, localError);
+            return;
+        }
+        
+        NSMutableDictionary *jsonDictionary = (NSMutableDictionary *)response;
+        
+        MSIDAADV2TokenResponse *tokenResponse = (MSIDAADV2TokenResponse *)[self.oauth2Factory tokenResponseFromJSON:jsonDictionary context:nil error:&error];
+        
+        if (!tokenResponse)
+        {
+            completionBlock(nil, [MSALErrorConverter MSALErrorFromMSIDError:error]);
+            return;
+        }
+        
+        if (![self.oauth2Factory verifyResponse:tokenResponse context:nil error:&error])
+        {
+            completionBlock(nil, [MSALErrorConverter MSALErrorFromMSIDError:error]);
+            return;
+        }
+        
+        if (_parameters.account.homeAccountId.identifier != nil &&
+            ![_parameters.account.homeAccountId.identifier isEqualToString:tokenResponse.clientInfo.accountIdentifier])
+        {
+            NSError *userMismatchError = CREATE_MSID_LOG_ERROR(_parameters, MSALErrorMismatchedUser, @"Different user was returned from the server");
+            completionBlock(nil, userMismatchError);
+            return;
+        }
+        
+        MSIDConfiguration *configuration = _parameters.msidConfiguration;
+        
+        BOOL isSaved = [self.tokenCache saveTokensWithConfiguration:configuration
+                                                           response:tokenResponse
+                                                            context:_parameters
+                                                              error:&error];
+        
+        if (!isSaved)
+        {
+            completionBlock(nil, [MSALErrorConverter MSALErrorFromMSIDError:error]);
+            return;
+        }
+        
+        MSIDAccessToken *accessToken = [self.oauth2Factory accessTokenFromResponse:tokenResponse configuration:configuration];
+        MSIDIdToken *idToken = [self.oauth2Factory idTokenFromResponse:tokenResponse configuration:configuration];
+        
+        MSALResult *result = [MSALResult resultWithAccessToken:accessToken idToken:idToken];
```

Comment:
nit: check that accessToken and idToken are not nil

Reply:
done